compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java

```diff
@@ -161,7 +161,7 @@ public static void markFieldsAsImmutable(BLangClassDefinition classDef, SymbolEn
         }
     }
 
-    private static BType getImmutableType(Location pos, Types types, BType type, SymbolEnv env,
+    public static BType getImmutableType(Location pos, Types types, BType type, SymbolEnv env,
```

Comment:
Since some arguments passed to this with the changes in this PR are still null, shall we introduce a separate overloaded method instead of making this public?

Reply:
Will address with this issue separately: https://github.com/ballerina-platform/ballerina-lang/issues/33598